tools/api_proto_plugin/type_context.py

```diff
@@ -60,22 +60,34 @@ def LocationPathLookup(self, path):
 
   # TODO(htuch): consider integrating comment lookup with overall
   # FileDescriptorProto, perhaps via two passes.
-  def LeadingCommentPathLookup(self, path):
+  def LeadingCommentPathLookup(self, path, annotation_xforms=None):
     """Lookup leading comment by path in SourceCodeInfo.
 
     Args:
       path: a list of path indexes as per
         https://github.com/google/protobuf/blob/a08b03d4c00a5793b88b494f672513f6ad46a681/src/google/protobuf/descriptor.proto#L717.
+      annotation_xforms: a dict of transformers to process annotation extracted
 
     Returns:
       Comment object.
     """
     location = self.LocationPathLookup(path)
-    if location is not None:
-      return Comment(
-          location.leading_comments,
-          annotations.ExtractAnnotations(location.leading_comments, self.file_level_annotations))
-    return Comment('', {})
+    if location is None:
```

Comment:
I'm not sure if this logic needs to live in the `TypeContext`. Could `LeadingCommentPathLookup` return an object that contains the comments, annotations, etc. and this object would have a way to do transforms?

Reply:
Sounds good.